isso/__init__.py

```diff
@@ -81,6 +81,11 @@
 
 logger = logging.getLogger("isso")
 
+def error_handler(env, request, error):
+    if "application/json" in request.headers["Accept"]:
```

Comment:
This should be a little bit more intelligent; i.e. if the preferred encoding is not json, it should not return json.

Reply:
What do you mean here @jelmer? `Accept-Encoding`?

Reply:
the accept header has a syntax, just doing a string search for "application/json" doesn't say whether that's actually the preferred encoding. What is the user sends "Accept: text/html, application/json" ?

See https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept

Reply:
I see. This would be an example header:
`Accept: text/html, application/xhtml+xml, application/xml;q=0.9, image/webp, */*;q=0.8`

Perhaps create iterable by splitting on comma (+strip whitespace yada yada), filter out everything except `text/html` and `application/json`, then return based on first match, else text-only response.